foundations/src/settings/mod.rs

```diff
@@ -310,6 +310,23 @@ use std::path::Path;
 /// }
 /// ```
 ///
+/// # Deny unknown fields
+///
+/// By default, this macro will automatically annotate types with `#[serde(deny_unknown_fields)]`.
+/// This means any keys in the config file that don't map directly to fields in the type will
+/// lead to a panic on deserialization.
```

Comment:
Not panic, but error. We should probably also explain motivation here: that it's usually desirable for services to use up to date configuration scheme and it's better for services to fail early on a bootstrap stage rather than run misconfigured.

Reply:
Good points, thank you.